src/Http/Owin/src/DictionaryStringValuesWrapper.cs

```diff
@@ -119,4 +121,40 @@ bool IDictionary<string, StringValues>.TryGetValue(string key, out StringValues
         value = default(StringValues);
         return false;
     }
+
+    public struct Enumerator : IEnumerator<KeyValuePair<string, StringValues>>, IEnumerator
```

Comment:
can this not share as `public struct Enumerator : IEnumerator<KeyValuePair<string, T>>, IEnumerator` with the one above? ConvertingEnumerator or something

Reply:
renamed to `ConvertingEnumerator`. I dont think I can make a generic impl for inner enumerator of `DictionaryStringArrayWrapper` and `DictionaryStringValuesWrapper`. That was your idea, right?

Reply:
You'd have to pass in a Convert delegate to reuse the same one. Maybe not worth the bother.